### PR TITLE
refactor(v2): init events

### PR DESCRIPTION
**Pull request by adlerhurst**
these events are used for later PRs.

This PR includes the changes of https://github.com/zitadel/zitadel/pull/7806 merge the successor first before you start reviewing this PR.

Part of https://github.com/zitadel/zitadel/issues/7639

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/v2/org/changed.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import "github.com/zitadel/zitadel/internal/v2/eventstore"
+
+var (
+	// TODO: use same logic as in [strings.Builder] to get rid of the following line
+	Changed ChangedEvent
+)
+
+type changedPayload struct {
+	Name *string `json:"name,omitempty"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/org/changed.go:11]: changed to string

File: internal/v2/org/state.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	if !s.IsValid() {
+		return false
+	}
+	for _, state := range states {
+		if s.Is(state) {
+			return true
+		}
+	}
+	return false
+}
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/org/state.go:36]: removed

File: internal/v2/user/human_registered.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	DisplayName       string              `json:"displayName,omitempty"`
+	PreferredLanguage language.Tag        `json:"preferredLanguage,omitempty"`
+	Gender            domain.Gender       `json:"gender,omitempty"`
+	EmailAddress      domain.EmailAddress `json:"email,omitempty"`
+	PhoneNumber       domain.PhoneNumber  `json:"phone,omitempty"`
+	Country           string              `json:"country,omitempty"`
+	Locality          string              `json:"locality,omitempty"`
+	PostalCode        string              `json:"postalCode,omitempty"`
+	Region            string              `json:"region,omitempty"`
+	StreetAddress     string              `json:"streetAddress,omitempty"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/user/human_registered.go:25]: removed

File: internal/v2/user/machine_secret_set.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package user
+
+import (
+	"github.com/zitadel/zitadel/internal/v2/eventstore"
+)
+
+type machineSecretSetPayload struct {
+	// New events only use EncodedHash. However, the ClientSecret field
+	// is preserved to handle events older than the switch to Passwap.
+	// ClientSecret *crypto.CryptoValue `json:"clientSecret,omitempty"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/user/machine_secret_set.go:10]: changed

File: internal/v2/user/human_init_code_added.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	"time"
+
+	"github.com/zitadel/zitadel/internal/v2/eventstore"
+)
+
+type humanInitialCodeAddedPayload struct {
+	Expiry            time.Duration `json:"expiry,omitempty"`
+	TriggeredAtOrigin string        `json:"triggerOrigin,omitempty"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/user/human_init_code_added.go:11]: changed

File: internal/v2/user/human_password_changed.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package user
+
+import (
+	"github.com/zitadel/zitadel/internal/v2/eventstore"
+)
+
+type humanPasswordChangedPayload struct {
+	// New events only use EncodedHash. However, the secret field
+	// is preserved to handle events older than the switch to Passwap.
+	// Secret            *crypto.CryptoValue `json:"secret,omitempty"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/user/human_password_changed.go:10]: changed

File: internal/v2/user/token_added.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+type tokenAddedPayload struct {
+	TokenID           string       `json:"tokenId"`
+	ApplicationID     string       `json:"applicationId"`
+	UserAgentID       string       `json:"userAgentId"`
+	RefreshTokenID    string       `json:"refreshTokenID,omitempty"`
+	Audience          []string     `json:"audience"`
+	Scopes            []string     `json:"scopes"`
+	Expiration        time.Time    `json:"expiration"`
+	PreferredLanguage language.Tag `json:"preferredLanguage"`
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/v2/user/token_added.go:19]: changed

**Review by livio-a — DISMISSED** [review]

**Review by livio-a — APPROVED** [review]

**Pull request merged**